modules/developers-guide/examples/javascript/node-full-script.mjs

```diff
@@ -0,0 +1,108 @@
+    const promisifiedClient = promisifyStargateClient(stargateClient);
+
+    console.log("promisified client")
+
+    // For Stargate OSS: generate authentication metadata that is passed in the executeQuery and executeBatch statements
+    const authenticationMetadata = await credentials.generateMetadata({service_url: auth_endpoint});
+
+    // For Stargate OSS: Create a new keyspace
+    const createKeyspaceStatement = new Query();
+    // Set the CQL statement
+    createKeyspaceStatement.setCql("CREATE KEYSPACE IF NOT EXISTS test WITH REPLICATION = {'class' : 'SimpleStrategy', 'replication_factor' : 1};");
+
+    await promisifiedClient.executeQuery(createKeyspaceStatement, authenticationMetadata);
+
+    console.log("created keyspace");
+
+    // For Stargate OSS: Create a new table
+    const createTableStatement = new Query();
+    // Set the CQL statement
+    createTableStatement.setCql("CREATE TABLE IF NOT EXISTS test.users (firstname text PRIMARY KEY, lastname text);");
+
+    await promisifiedClient.executeQuery(
+      createTableStatement,
+      authenticationMetadata
+    );
+
+    console.log("created table");
+
+    // For Stargate OSS: INSERT two rows/records
+    // Create two queries that will be run in a batch statement
+    const insertOne = new BatchQuery();
+    const insertTwo = new BatchQuery();
+
+    // Set the CQL statement
+    insertOne.setCql(`INSERT INTO test.users (firstname, lastname) VALUES('Lorina', 'Poland')`);
+    insertTwo.setCql(`INSERT INTO test.users (firstname, lastname) VALUES('Doug', 'Wettlaufer')`);
+
+    // Define the new batch to include the 2 insertions
+    const batch = new Batch();
+    batch.setQueriesList([insertOne, insertTwo]);
+
+    // For Stargate OSS: execute the batch statement
+    const batchResult = await promisifiedClient.executeBatch(
+      batch,
+      authenticationMetadata
+    );
+    console.log("inserted data");
+
+    // For Stargate OSS: SELECT the data to read from the table
+    const query = new Query();
+    const queryString = 'SELECT firstname, lastname FROM test.users;'
+    // Set the CQL statement using the string defined in the last line
+    query.setCql(queryString);
+
+    // For Stargate OSS and Astra DB: execute the query statement
+    const response = await promisifiedClient.executeQuery(
+      query,
+      authenticationMetadata
+    );
+
+    console.log("select executed")
+
+    // Get the results from the execute query statement
+    // and separate into an array to print out the results
+    const resultSet = response.getResultSet();
+    const rows = resultSet.getRowsList();
+
+    // This for loop gets 2 results
+    for ( let i = 0; i < 2; i++) {
+      var valueToPrint = "";
+      for ( let j = 0; j < 2; j++) {
+        var value = rows[i].getValuesList()[j].getString();
+        valueToPrint += value;
+        valueToPrint += " ";
+      }
+      console.log(valueToPrint);
+    }
```

Comment:
Nit: JS has some cool iteration functions that let you skip the formal for loop. And we can use [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) to skip the string concatenation:

```
rows.forEach(row => {
  const valueList = row.getValuesList();
  const firstName = valueList[0].getString();
  const lastName = valueList[1].getString();
  console.log(`${firstName} ${lastName}`);
}
```

Reply:
Gotcha! I was modifying code from the go example, and used the same pattern. If you test and add to the grpc-node-client repo, I'll modify the docs. :-)